virtcontainers/mount.go

```diff
@@ -281,6 +281,11 @@ type Mount struct {
 
 	// ReadOnly specifies if the mount should be read only or not
 	ReadOnly bool
+
+	// BlockDevice represents block device that is attached to the
+	// VM in case this mount is a block device file or a directory
+	// backed by a block device.
+	BlockDevice *BlockDevice
```

Comment:
Just wondering if a volume device is always gonna be a block device ? Or can it be something else ?

Reply:
You didn't answer this question.

Reply:
Volumes will mostly be a block based device, I suppose you can pass a character device here, but then we dont do any special handling, just pass it through 9p.

Reply:
In runv, we also support NFS volumes and ceph rbd volumes. I guess we can add a structure similar to agent's storage that can describe them all. But it does not block this PR and can be left for future improvements.